src/savegame.cpp

```diff
@@ -0,0 +1,197 @@
+#include <stdexcept>
+#include "savegame.h"
+
+Savegame::Savegame(const char *fileName) {
+    if (!_file.open(fileName)) {
+        throw std::runtime_error("Cannot open savegame file");
+    }
+}
+GameState* Savegame::load() {
+    GameState* gameState = new GameState;
+
+    readConfig(_file, gameState);
+    readGalaxy(_file, gameState);
+    readHeroes(_file, gameState);
+    readPlayers(_file, gameState);
+    readStars(_file, gameState);
+
+    return gameState;
+}
+
+Savegame::~Savegame() {
+    _file.close();
+}
+
+void Savegame::readConfig(File& file, GameState* state) {
+    file.seek(0x04, SEEK_SET);
+    file.read(state->_gameConfig.saveGameName, 23);
+
+    file.seek(0x29, SEEK_SET);
+    state->_gameConfig.stardate = file.readUint16LE();
+
+    file.seek(0x2E, SEEK_SET);
+    state->_gameConfig.endOfTurnSummary = file.readUint8();
+    state->_gameConfig.endOfTurnWait = file.readUint8();
+    state->_gameConfig.randomEvents = file.readUint8();
+    state->_gameConfig.enemyMoves = file.readUint8();
+    state->_gameConfig.expandingHelp = file.readUint8();
+    state->_gameConfig.autoSelectShips = file.readUint8();
+    state->_gameConfig.animations = file.readUint8();
+    state->_gameConfig.autoSelectColony = file.readUint8();
+    state->_gameConfig.showRelocationLines = file.readUint8();
+    state->_gameConfig.showGNNReport = file.readUint8();
+    state->_gameConfig.autoDeleteTradeGoodHousing = file.readUint8();
+    state->_gameConfig.showOnlySeriousTurnSummary = file.readUint8();
+    state->_gameConfig.shipInitiative = file.readUint8();
+}
+
+void Savegame::readGalaxy(File& file, GameState* state) {
+    file.seek(0x31be4, SEEK_SET);
+    state->_galaxy.sizeFactor = file.readUint8();
+    file.seek(0x31be9, SEEK_SET);
```

Comment:
`readUint32LE()` would work just as well and it'd be more clear how big the gap is.